### PR TITLE
Add support for scripted fields

### PR DESCRIPTION
**Pull request by CGamesPlay**
This change enables [scripted fields](https://www.elastic.co/guide/en/elasticsearch/reference/7.11/search-fields.html#script-fields) to be queried when examining ElasticSearch results. It does so by merging the `_source` result property with the `fields` result property for ElasticSearch documents.

I'm not totally thrilled with the implementation, because the `fieldsAttrName` method isn't completely accurate. However, at least within splainer, this method isn't even used outside of the `fieldsProperty` method. I'm open to feedback about the implementation choice, potentially even removing `fieldsAttrName` if this is bothersome.

### CODE REVIEW

**Comment by epugh** [comment]
@CGamesPlay I am horribly embarrassed that I for some reason didn't see this PR!   

I'm not super expert on ES, so I'll need someone else to weigh in on this.  Maybe @nathancday or @worleydl have an opinon on how this was done.

However, would it be possible to add a unit test demonstrating how this works?

File: factories/docFactory.js [hunk 1]
```diff
@@ -8,55 +8,65 @@
...
 
-      self.doc             = doc;
-
-      self.groupedBy       = groupedBy;
-      self.group           = group;
-      self.options         = options;
-      self.version         = version;
-      self.fieldsAttrName  = fieldsAttrName;
-      self.fieldsProperty  = fieldsProperty;
+    Doc.prototype = {};
```
Comment (epugh) [inline factories/docFactory.js:17]: Can you explain why this change w as made?  Is htis a "better" way to  do this?   Are there other places through the codebase that should be like this?

Comment (CGamesPlay) [reply factories/docFactory.js:17]: This change was made so that the methods can be overridden in subclasses. This is a technique called "prototype inheritance", and yes, it should be used everywhere on all "classes" in JavaScript. I believe there are other places in the code base where we already store the methods on the prototype (instead of creating new methods on each instantiated object). However, if you want to "modernize" this, you should instead migrate the code to [ES6 classes](https://caniuse.com/es6-class).

**Review by CGamesPlay — COMMENTED** [review]
Unfortunately, I don't have the bandwidth to write a unit test for this (we've already used it on our fork and are working on other projects now). Hopefully it isn't too difficult to add one, it's really just a matter of understanding that scripted fields occur in a different place in the result JSON from Elasticsearch than other fields do.

File: factories/docFactory.js (hunk 1 above)
Comment (epugh) [reply factories/docFactory.js:17]: Thanks for sharing this!   I think I'd like to seperate what you've provided into two PR's if I can..  the first is for the support for scripted fields, and the second for resolving this "prototype inheritance" issue..

**Comment by epugh** [comment]
Humm..  rethinking my earlier comment and will take the refactorings that you did, because that appears to be in line with the dominant pattern in `splainer-search`.   Still working on understanding enough to code a unit test!
